File: java/leetcode/single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int previousVal = nums[0];
        int foundElem = previousVal;
            for (int i = 0; i < nums.length; i++) {
                if (previousVal != nums[i] && (i + 1 == nums.length || nums[i] != nums[i + 1])) {
                    foundElem = nums[i];
                    break;
		}
                previousVal = nums[i];
        }
        return foundElem;
    }
}

File: python/leetcode/single-element-in-a-sorted-array.py
class Solution:
    def singleNonDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        previous_val = nums[1]
        found_elem = None
        for i in range(len(nums)):
            if previous_val != nums[i] and (i + 1 == len(nums) or nums[i] != nums[i + 1]):
                found_elem = nums[i]
                break
            previous_val = nums[i]
        return found_elem
